addons/im_livechat/static/tests/embed/livechat_session.test.js

```diff
@@ -76,15 +69,14 @@ test("Seen message is saved on the server", async () => {
     await loadDefaultEmbedConfig();
     const userId = serverState.userId;
     await start({ authenticateAs: false });
-    await mountWithCleanup(LivechatButton);
     await click(".o-livechat-LivechatButton");
     await contains(".o-mail-Thread");
     await insertText(".o-mail-Composer-input", "Hello, I need help!");
     triggerHotkey("Enter");
     await contains(".o-mail-Message", { text: "Hello, I need help!" });
     await waitUntilSubscribe();
-    const initialSeenMessageId =
-        getService("im_livechat.livechat").thread.selfMember.seen_message_id?.id;
+    const initialSeenMessageId = Object.values(getService("mail.store").Thread.records).at(-1)
+        .selfMember.seen_message_id?.id;
```

Comment:
This is weak against change in data and ordering of threads that could change.

Ok for now, but IMO I regret we don't have `data-thread-id`/`data-thread-model` in the template of `Thread` component. While this looks useless in practice, this makes debugging easier and could also help tests to be more robust in introspecting stuffs in JS models like here.

Reply:
I agree, these new hacked id will be annoying to cover to tours and to maintain generally.
Considering the number of occurrences we should find something.
Why do we need id in the first place?
The problem is that we are manually doing stuff rather than relying on actual flow/route/method.